Add `encoding` selector option (base64 or binary) (#39)
logstash-plugins/logstash-codec-avro: docs/index.asciidoc

Controls this plugin's compatibility with the {ecs-ref}[Elastic Common Schema (ECS)].

[id="plugins-{type}s-{plugin}-base64_encoding"]

> ===== `base64_encoding`

Reviewer: I would prefer we avoid adding boolean options tied to the current implementation detail of there being two options, instead opting for a name that is descriptive of what the option controls and a value that is descriptive of what is being done. This allows us to leave room for additional options in the future, such as alternate encodings that may be introduced upstream.

In this case, something like:

`encoding` with values `base64` and `binary`.

Reviewer 2: +1 on previous comment.  If you take this suggestion, the doc will need to be reworded. Please ping me and I'll review/retest doc then.   :-)

Author: Thanks for suggesting it. I've applied the hint, so it's ready for a review :-)

* Value can be any of: `binary`, `base64`
* Default value is `base64`

Set encoding for Avro's payload. 

> Use `base64` (default) encoding to convert the raw binary bytes to a `base64` encoded string.
> 
> Set this option to `binary` to use the plain binary bytes.

Reviewer: ```suggestion
Use `base64` (default) to indicate that this codec sends or expects to receive base64-encoded bytes.

Set this option to `binary` to indicate that this codec sends or expects to receive binary Avro data.

```

> Select which encoding to use to represent Avro's payload.
> By default, it uses `base64`, however could be customized with `binary` to use the plain binary bytes.
> With `base64` encoding the raw binary bytes are converted to a `base64` encoded string.
> 

Reviewer 2: ```suggestion
Set encoding for Avro's payload. 
Use `base64` (default) encoding to convert the raw binary bytes to a `base64` encoded string.

Set this option to `binary` to use the plain binary bytes.
```

Author: Thank's @karenzone accepted your wonderful suggestion! :-)
